src/thought/view-thoughts/thoughts-mind-map-view.js

```diff
@@ -1,281 +1,286 @@
+import React, {Component} from 'react';
+
 import BrainVisNetwork from 'thought/view-thoughts/brain-vis-network';
 import VisNetworkHelper from 'thought/view-thoughts/vis-network-helper';
 import thoughtStorage from 'storage/thought-storage';
 import siteGlobalLoadingBar from 'ui/spinner/site-global-loading-bar';
 import _ from 'underscore';
-import thoughtNameEditor from 'ui/thought-name-editor';
-import thoughtContentEditor from 'ui/thought-content-editor';
+import {ThoughtNameEditor} from 'ui/thought-name-editor';
 
 let brainVisNetwork;
 let visNetworkHelper;
-let thoughts = [];
-let currentViewedThought;
-let currentViewedThoughtId;
-let initialThought;
 
 let spinner = siteGlobalLoadingBar.create('mind map');
 
-export default {render};
-
-/**
- * Set thoughts and selectedThought.
- * @param {Array} options.thoughts - Array of thoughts to render.
- * @param {String} options.selectedThoughtId - Id of thought that should be in focus (center).
- */
-function setOptions(options) {
-  console.debug('thoughtsMindMapView.setOptions(), options: ', options);
-  thoughts = options.thoughts;
-  console.debug('options.selectedThoughtId: ', options.selectedThoughtId);
-  initialThought = options.selectedThought;
-  currentViewedThought = initialThought;
-  currentViewedThoughtId = initialThought.id;
-  console.debug('currentViewedThoughtId: ', currentViewedThoughtId);
-}
+export class ThoughtsMindMapView extends Component {
+  state = {editNote: null};
 
-function render(options) {
-  setOptions(options);
+  ref = React.createRef();
 
-  let element = document.createElement('div');
-  element.id = 'thoughts-container';
+  render() {
+    const {editNote} = this.state;
 
-  brainVisNetwork = new BrainVisNetwork({
-    containerDomElement: element
-  });
+    return (
+      <div ref={this.ref} id="thoughts-container">
+        {editNote && <ThoughtNameEditor
+          thought={thoughtStorage.findThoughtById(editNote.id)}
```

Comment:
editing is always for a selected note, so we could use `selectedNote.id` here.

Or if we want a separate variable for this, we could name it `noteToEdit` or `noteBeingEdited`.

Reply:
renamed to "selectedNote"